test/rollup.js

```diff
@@ -20,10 +20,11 @@ test('xdm (rollup)', async function (t) {
     plugins: [
       {
         name: 'xdm',
+        // @ts-ignore TODO narrow the types of file.contents and file.map
         async transform(contents, filePath) {
           if (path.extname(filePath) !== '.mdx') return null
           var file = await compile({contents, path: filePath})
-          return {code: String(file.contents), map: file.map?.toString()}
+          return {code: file.contents, map: file.map}
```

Comment:
here’s an example that shows it works: https://github.com/rollup/plugins/blob/f4999af9ebcc495a0a0f43aeba0c3cc820a58ee6/packages/dsv/src/index.js#L29

Reply:
right, the issue here is that `VFile` has `contents` as being `string | Uint8Array` and `map` being `unknown` since it is not a standard `VFile` property.
To make this seemless in typescript both of these would need to be narrowed to be compatible with the rollup type.

Reply:
Oh, I’m fine with `String(file.contents)` btw, I think my problem was with `map: file.map.toString()` (because that would yield `'undefined'` or `[object Object]`)

Reply:
:man_shrugging: if you prefer the cast I can add it back, the type conflict with `map` would remain, so either way there will be a `@ts-ignore` until this can be refined.

Reply:
I have no preference for `file.contents`, whatever works for Rollup/TS!

Reply:
keeping as it originally was for now, may revisit down the road.